src/main/java/io/wisetime/connector/IntegrateApplication.java

```diff
@@ -57,6 +65,10 @@ private void addEndpoints() {
 
     post("/receiveTimePostedEvent", (request, response) -> {
       TimeGroup userPostedTime = om.readValue(request.body(), TimeGroup.class);
+      if (!apiKey.equals(request.headers("x-api-key"))) {
```

Comment:
@vyshane 
We had a discussion about web hook security in slack earlier this week.
I understood it, that you were saying to use the api-key as a tool to verify that the posted time was sent by us to the web hook. Was that a misunderstanding?

If the callerKey field should be used: What should it contain and how should it be shared with the client? Also the api key for now until we are ready to go with the JWT?

In my eyes: I think it would be better, when we want to switch to JWT in the future, to reserve the callerKey field for it and not switch content of the field to another auth method.

Reply:
I think there was a miscommunication on this. Looks like I managed to confuse matters. Sorry about that.

The `callerKey` is provided to us by the connector when they register their webhook: https://wisetime.io/docs/connect/api/#operation/post-subscribe

We send it back to the connector when we call the webhook, closing the loop so to speak.

API key is provided by us to connector; connector sends it back to us to authenticate their requests
Caller key is provided by connector to us; we send it back to them to authenticate our requests